File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        int num = 1;
        
        while(n > num){
            answer++;
            num *= answer;
        }
        
        answer = n == num ? answer : answer - 1;
        
        return answer;
    }
    
}

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    int temp = 1;
    
    while(temp <= n){
        answer++;
        temp *= answer;
    }
    
    answer--;
    
    return answer;
}